packages/lodestar/src/chain/archiver/index.ts

```diff
@@ -65,24 +65,52 @@ export class Archiver {
   };
 
   private processFinalizedCheckpoint = async (finalized: CheckpointWithHex): Promise<void> => {
+    const logError = (e: Error): void => {
+      this.logger.error("Error processing finalized checkpoint", {epoch: finalized.epoch}, e);
+    };
     try {
       const finalizedEpoch = finalized.epoch;
       this.logger.verbose("Start processing finalized checkpoint", {epoch: finalizedEpoch});
       await archiveBlocks(this.db, this.chain.forkChoice, this.chain.lightClientServer, this.logger, finalized);
 
-      // should be after ArchiveBlocksTask to handle restart cleanly
-      await this.statesArchiver.maybeArchiveState(finalized, this.chain.anchorSlot);
-
-      await Promise.all([
-        this.chain.checkpointStateCache.pruneFinalized(finalizedEpoch),
-        this.chain.stateCache.deleteAllBeforeEpoch(finalizedEpoch),
-      ]);
-
       // tasks rely on extended fork choice
       this.chain.forkChoice.prune(finalized.rootHex);
+
+      // The following tasks need not be awaited and can run independently
+      // should be after ArchiveBlocksTask to handle restart cleanly
+      this.statesArchiver.maybeArchiveState(finalized).catch(logError);
+      this.updateBackfillRange(finalized).catch(logError);
+      this.chain.stateCache.deleteAllBeforeEpoch(finalizedEpoch).catch(logError);
+      this.chain.checkpointStateCache.pruneFinalized(finalizedEpoch).catch(logError);
```

Comment:
Restore previous sequence flow, maybeArchiveState is fully completed before pruning states

Reply:
Do you have any reason to believe that this prune functions will error very frequently? Why do they have dedicated error handling?

Reply:
Given that `this.updateBackfillRange(finalized).catch(logError);` is the least critical step it should be done last

Reply:
> Do you have any reason to believe that this prune functions will error very frequently? Why do they have dedicated error handling?

If no, just use the original single try / catch without any additional .catch(). If latter when running the node data shows that some of this statements error often we can add additional try / catch blocks

Reply:
done! :+1: thank you for pointing out the improper flow :heart: